numpy/lib/recfunctions.py

```diff
@@ -1069,7 +1069,7 @@ def unstructured_to_structured(arr, dtype=None, names=None, align=False,
 
     # first view as a packed structured array of one dtype
     packed_fields = np.dtype({'names': names,
-                              'formats': [(arr.dtype, c) for c in counts]})
+                              'formats': [(arr.dtype, (c,)) for c in counts]})
```

Comment:
To work around #13327 you need to special-case `c == 1` here

Reply:
Nevermind - the actual problem is the test is bad - it should be checking `ret.base is a or ret is a` - or in a shorter form, `ret.view().base is a`

Reply:
Switching to either form still fails for me locally.